### PR TITLE
docs: docs for AVRO and JSON_SR keys

### PR DESCRIPTION
**Pull request by vcrfxia**
### Description 

Docs to accompany https://github.com/confluentinc/ksql/pull/6694

### Testing done 

Docs-only change.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/concepts/schemas.md [hunk 1]
```diff
@@ -141,6 +136,27 @@ time the statement is first executed.
...
 #### With a key column
 
+The following statement shows how to create a new `pageviews` stream by reading
+from a {{ site.ak }} topic that has Avro-formatted key and message values.
+
+```sql
+CREATE STREAM pageviews WITH (
+    KAFKA_TOPIC='pageviews-avro-topic',
+    KEY_FORMAT='AVRO',
+    VALUE_FORMAT='AVRO'
```
Comment (agavra) [inline docs/concepts/schemas.md:146]: should we also mention `FORMAT='AVRO'`? I suspect this might be somewhat common

**Review by agavra — APPROVED** [review]
Thanks for getting ahead on the docs!

File: docs/concepts/schemas.md [hunk 2]
```diff
@@ -77,30 +77,25 @@ ksqlDB is [configured to use it](../operate-and-deploy/installation/server-confi
...
+    Message *keys* in Protobuf are not supported. If your message keys
     are in an unsupported format, see [What to do if your key is not set or is in a different format](../developer-guide/syntax-reference.md#what-to-do-if-your-key-is-not-set-or-is-in-a-different-format). 
-    JSON message keys can be accessed by defining the key as a single `STRING` value, which will 
-    contain the JSON document.
     
-Although ksqlDB doesn't support loading the message key's schema from {{ site.sr }},
-you can provide the key column definition within the `CREATE TABLE` or `CREATE STREAM`
-statement, if the data records are compatible with ksqlDB. This is known as
-_partial schema inference_, because the key schema is provided explicitly.
+If declaring a stream or table with a key format that is different from its
```
Comment (JimGalasyn) [inline docs/concepts/schemas.md:93]:
```suggestion
If you're declaring a stream or table with a key format that's different from its

```

File: docs/concepts/schemas.md [hunk 3]
```diff
@@ -77,30 +77,25 @@ ksqlDB is [configured to use it](../operate-and-deploy/installation/server-confi
...
+value format, and only one of the two formats supports schema inference,
+you can explicitly provide the columns for the format that does not support schema inference
+while still having ksqlDB load columns for the format that does support schema inference
+from m {{ site.sr }}. This is known as _partial schema inference_. To infer value columns
```
Comment (JimGalasyn) [inline docs/concepts/schemas.md:97]:
```suggestion
from {{ site.sr }}. This is known as _partial schema inference_. To infer value columns

```

**Review by JimGalasyn — APPROVED** [review]
Looks awesome! Just a couple of copy edits.

File: docs/concepts/schemas.md (hunk 1 above)
Comment (vcrfxia) [reply docs/concepts/schemas.md:146]: Were you imagining a tip saying that this is equivalent to simply specifying `FORMAT`, updating these examples to use `FORMAT`, or something else? The `FORMAT` property is documented in https://github.com/confluentinc/ksql/blob/master/docs/developer-guide/ksqldb-reference/create-stream.md (and similar for CT, CSAS, and CTAS) which feels like the more appropriate place to formally introduce it, but I can certainly slip in a mention of it here as well.

Comment (vcrfxia) [reply docs/concepts/schemas.md:146]: Discussed offline -- leaving this as is for now. Will see if we can consolidate information about keys into a natural place in a subsequent PR.

**Pull request merged**